src/test/scala/com/sky/kafka/configurator/KafkaTopicAdminSpec.scala

```diff
@@ -28,31 +28,17 @@ class KafkaAdminClientSpec extends KafkaIntSpec with Eventually {
     }
   }
 
-  it should "return a failure if given an unrecognised config entry" in {
-    val inputTopic = someTopic.copy(config = Map(
-      "invalid.key" -> "invalid.value"
-    ))
-
-    adminClient.create(inputTopic) shouldBe a[Failure[_]]
-  }
-
-  it should "return a failure if replication factor is greater than number of available brokers" in {
-    val inputTopic = someTopic.copy(replicationFactor = 2)
-
-    adminClient.create(inputTopic) shouldBe a[Failure[_]]
-  }
-
```

Comment:
The removal of these two test is a concern for me... since it does change slightly the behaviour of the application, and doesn't error where it was erring before... I wasn't to worried since as before it does what the underlying `kafka-clients` library does, however it is worthy for someone with more insight in `kafka-configurator` to consider whether we should honour this test or it is good to not to fail in this cases...

Thoughts?

Reply:
as discussed yesterday, I am a bit worried about this too. What would keeping this behaviour involve now that the underlying client behaves differently?

Reply:
the question is really what does this behaviour gives us? what do we loose by not having it?

Reply:
its just so that the user knows whether their topics were successfully configured or not. 

I would expect, as a user, that if i don't see anything that says otherwise and the exit code is 0 then all of my topics were configured successfully and the YML file I provided can now be viewed as a source of truth for the state of the topics listed.

What actually happens here? if i submit a topic config for a new topic, and the replication factor is higher than the number of available brokers, will my topic be created at all? or will it just not create the topic?

Reply:
Ok, I think we do agree that something needs to be done, so it does fail when the change is effectively not applied.

I propose to do a fetch after applying the change and error if the change has not had any effect. I also propose on amending how the Topics are compared when deciding whether to apply the change or not.

Thoughts?

Reply:
"I propose to do a fetch after applying the change and error if the change has not had any effect."

Sounds good.

"I also propose on amending how the Topics are compared when deciding whether to apply the change or not."

In what way?